File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        l = 1
        r = [nums[-1]] * len(nums)
        for i in range(len(nums)-2, -1, -1):
            r[i] = r[i+1] * nums[i]
        res = []

        for i in range(len(nums)):
            if i == 0: res.append(r[i+1])
            elif i == len(nums) - 1: res.append(l)
            else:
                res.append(l * r[i+1])
            l *= nums[i]
        
        return res

File: Data Structures & Algorithms/concatenation-of-array/submission-3.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        vector<int> v(nums.size() * 2, 0);

        for (int i = 0; i < nums.size(); i ++) {
            v[i] = nums[i];
            v[i + nums.size()] = nums[i];
        }

        return v;
    }
};